feat(godaddy): add option to disable api key/secret validation

alexstojda opened the pull request:
## What does it do ?

The previous change I made did not fix the rate limiting issues. This will provide an option to just skip the validation all together.

Flag will default to false, so that existing installations will still validate the API Key/Secret

## Motivation

> During startup, the client calls the endpoint https://api.godaddy.com/v1/domains?statuses=ACTIVE,PENDING_DNS_ACTIVE1 to test and validate that the credentials are working. 

#4866 sought to fix an underlying issue by ignoring QUOTA_EXCEEDED errors on the initial validation call, since theory was that there was a per-endpoint limit. However, after running the fix for the last year, we still hit some other rate limit fairly often. 

My understanding now is that there is still some global & account-level rate limit whose threshold is often hit because of the initial "validation" request. 

Considering there are very few people reporting that they are running into this issue (I hypothesize this is because not many people are using the godaddy provider at a similar scale), I decided that it would be a more effective approach to provide the option to disable that validation on startup altogether.

## More

- [x] Yes, this PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Yes, I added unit tests
- [x] Yes, I updated end user documentation accordingly

<!--
    Please read https://github.com/kubernetes-sigs/external-dns#contributing before submitting
    your pull request. Please fill in each section below to help us better prioritize your pull request. Thanks!
-->

Relates to #4866

alexstojda committed:
docs: update flags documentation

mloiseleur commented:
Can you detail your motivation?

:thinking: How avoiding validation on the API key may fix the rate limit issue with the GoDaddy API?

mloiseleur commented:
@alexstojda I'm probably missing something here :thinking:. 

Are you saying that you hit the rate/limit because of

a. a _single_ startup request?
b. a request that is launched on every reconcile loop?

alexstojda committed:
feat(godaddy): add option to disable api key/secret validation

The previous change I made did not fix the rate limiting issues. This will provide an option to just skip the validation all together.

Flag will default to false, so that existing installations will still validate the API Key/Secret

# Conflicts:
#	pkg/apis/externaldns/types.go

alexstojda committed:
fix(godaddy): rename flag and clarify logic

# Conflicts:
#	pkg/apis/externaldns/types.go

ivankatliarchuk commented:
As alternative worth to add HTTP rate limit (similar request for another provicer https://github.com/kubernetes-sigs/external-dns/issues/4572) + zone caching(https://github.com/kubernetes-sigs/external-dns/blob/bb67452f3853170ae26d8a5b9fac0e96fb8e614c/provider/azure/cache.go#L24) + provider caching (https://github.com/kubernetes-sigs/external-dns/blob/master/provider/cached_provider.go)

ivankatliarchuk commented:
/close
